pkg/entities/cache.go

```diff
@@ -0,0 +1,59 @@
+package entities
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/defipod/mochi/pkg/request"
+	"github.com/defipod/mochi/pkg/response"
+)
+
+func (e *Entity) SetUpvoteMessageCache(req *request.SetUpvoteMessageCacheRequest) error {
+	// "upvote_msg_<userID>": "<messageID>_<channelID>_<guildID>"
+
+	key := fmt.Sprintf("upvote_msg_%s", req.UserID)
+	value := fmt.Sprintf("%s_%s_%s", req.MessageID, req.ChannelID, req.GuildID)
+	err := e.cache.Set(key, value, time.Duration(time.Minute*5))
+	if err != nil {
+		e.log.Errorf(err, "[e.SetUpvoteMessageCache] failed to set cache")
```

Comment:
e.log.Fields(logger.Fields{"key": key, "value": value}).Error(...)
With this can monitor which key error